### PR TITLE
Add a developer mode configuration section

### PR DESCRIPTION
**Pull request by bparees**
## Description

Introduce a new config section for developer-only settings.  Initial settings:

set llm generation temperature
enable/disable ui
enable/disable question validation
enable/disable auth (not wired to anything yet)

## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue #
- Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

**Comment by bparees** [comment]
/assign @syedriko 

@syedriko ptal as our current "config guru"

File: ols/app/models/config.py [hunk 1]
```diff
@@ -383,11 +380,59 @@ def validate_yaml(self) -> None:
         self.logging_config.validate_yaml()
 
 
+class DevConfig(BaseModel):
+    """Developer-mode-only configuration options."""
+
+    enable_dev_ui: bool = False
+    disable_question_validation: bool = False
+    llm_temperature_override: int | None = None
```
Comment (syedriko) [inline ols/app/models/config.py:388]: Should llm_temperature_override be a float?

Comment (bparees) [reply ols/app/models/config.py:388]: apparently so, thanks :facepalm:

File: ols/app/models/config.py [hunk 2]
```diff
@@ -418,6 +466,8 @@ def validate_yaml(self) -> None:
         if self.ols_config is None:
             raise InvalidConfigurationError("no OLSConfig found")
         self.ols_config.validate_yaml()
+        if self.dev_config:
```
Comment (syedriko) [inline ols/app/models/config.py:469]: Seems the `if` is not needed if dev_config is always there. May be it doesn't need to be Optional?

Comment (bparees) [reply ols/app/models/config.py:469]: yeah this was a leftover from when i originally had dev_config as possible to be None, thanks for catching it.

File: ols/app/models/config.py [hunk 3]
```diff
@@ -383,11 +380,59 @@ def validate_yaml(self) -> None:
...
+    def __init__(self, data: Optional[dict] = None):
+        """Initialize developer configuration settings."""
+        super().__init__()
+        if data is None:
+            return
+        self.enable_dev_ui = str(data.get("enable_dev_ui", "False")).lower() == "true"
+        self.disable_question_validation = (
+            str(data.get("disable_question_validation", "False")).lower() == "true"
+        )
+        self.llm_temperature_override = data.get("llm_temperature_override", None)
```
Comment (syedriko) [inline ols/app/models/config.py:402]: FWIW, here's a pattern I'm using for getting an int out of YAML: https://github.com/syedriko/lightspeed-service/blob/77970fd8a235dfa25e4ff1fd3d926214472463be/ols/app/models/config.py#L191

File: ols/app/models/config.py (hunk 2 above)
Comment (bparees) [reply ols/app/models/config.py]: fixed

File: ols/app/models/config.py (hunk 1 above)
Comment (bparees) [reply ols/app/models/config.py]: fixed

File: ols/app/models/config.py (hunk 3 above)
Comment (bparees) [reply ols/app/models/config.py:402]: this feels like a situation where it's 6 one way and half a dozen the other  :shrug:

File: ols/app/models/config.py [hunk 4]
```diff
@@ -383,11 +380,59 @@ def validate_yaml(self) -> None:
...
+    def validate_yaml(self) -> None:
+        """Validate OLS Dev config."""
+        if self.llm_temperature_override is not None:
+            if not isinstance(self.llm_temperature_override, (float, int)):
+                raise InvalidConfigurationError(
+                    "llm_temperature_override must be a float"
+                )
+            if self.llm_temperature_override < 0 or self.llm_temperature_override > 1:
+                raise InvalidConfigurationError(
+                    "llm_temperature_override must be between 0 and 100"
```
Comment (syedriko) [inline ols/app/models/config.py:426]: between 0 and 1?

Comment (bparees) [reply ols/app/models/config.py]: the knock on effects from one change......

fixed.

**Pull request merged**